airflow/models/dataset.py

```diff
@@ -76,7 +76,9 @@ class DatasetModel(Base):
 
     @classmethod
     def from_public(cls, obj: Dataset) -> DatasetModel:
```

Comment:
I would probably say lets remove this function from this class entirely and move it to on the Manager instead -- that also might give us a better name than `resolve`.

How about `model = manager.from_public(dataset)` or `model = manager.make_dataset_model(dataset)`